lib/rust/performance-logger/src/lib.rs

```diff
@@ -0,0 +1,395 @@
+            let measurement: Result<Measurement, _> = entry.try_into();
+            if let Ok(measurement) = measurement {
+                measures.push(measurement)
+            }
+        });
+        Report { measures }
+    }
+}
+
+impl Display for Report {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let sorted_measurements =
+            self.measures.iter().sorted_by_key(|measure| OrderedFloat(measure.start_time));
+        for to_print in sorted_measurements {
+            f.write_str(&format!("{}\n", &to_print))?;
+        }
+        Ok(())
+    }
+}
+
+
+
+// ===============================
+// === Convenience Definitions ===
+// ===============================
+
+macro_rules! define_start_interval_helpers {
+    ($log_level:expr, $log_level_name:ident) => {
+        paste::paste! {
+            #[doc = "Start measuring an interval with log level `" $log_level_name "`. See `start_interval` for more information."]
+            pub fn  [< start_interval_ $log_level_name >](interval_name: &str) -> IntervalHandle {
+                start_interval($log_level, interval_name)
+            }
+        }
+    };
+}
+
+define_start_interval_helpers!(LogLevel::SECTION, section);
+define_start_interval_helpers!(LogLevel::TASK, task);
+define_start_interval_helpers!(LogLevel::DETAIL, detail);
+define_start_interval_helpers!(LogLevel::DEBUG, debug);
+
+
+macro_rules! define_end_interval_helpers {
+    ($log_level:expr, $log_level_name:ident) => {
+        paste::paste! {
+            #[doc = "End measuring an interval with log level `" $log_level_name "`. See `end_interval` for more information."]
+            pub fn  [< end_interval_ $log_level_name >](interval_name: &str) {
+                if let Err(error) = end_interval($log_level, interval_name) {
+                    WARNING!(format!("Performance logging encountered an error when ending an interval: {:?}", error));
+                };
+            }
+        }
+    };
+}
+
+define_end_interval_helpers!(LogLevel::SECTION, section);
+define_end_interval_helpers!(LogLevel::TASK, task);
+define_end_interval_helpers!(LogLevel::DETAIL, detail);
+define_end_interval_helpers!(LogLevel::DEBUG, debug);
+
+
+macro_rules! define_measure_interval_helpers {
+    ($log_level:expr, $log_level_name:ident) => {
+        paste::paste! {
+            #[doc = "Measure the execution time of the given interval with log level `" $log_level_name  "`. See `measure_interval` for more information."]
+            pub fn [< measure_interval_ $log_level_name >]<T, F: FnMut() -> T>(interval_name: &str, closure: F) -> IntervalMeasurementResult<T>  {
+                measure_interval($log_level, interval_name, closure)
+            }
+        }
+    };
+}
+
+
+define_measure_interval_helpers!(LogLevel::SECTION, section);
+define_measure_interval_helpers!(LogLevel::TASK, task);
+define_measure_interval_helpers!(LogLevel::DETAIL, detail);
+define_measure_interval_helpers!(LogLevel::DEBUG, debug);
```

Comment:
There are no tests showing the usage of this and no example scene in Rust that we can run to check if it works.

Reply:
There is some example usage placed in `app/gui/src/ide/integration/project.rs` as well as examples of the JS API in `app/ide-desktop/lib/content/src/index.ts`.